tools/gfx/render.h

```diff
@@ -860,223 +1083,229 @@ class IRenderer: public ISlangUnknown
     {
         int width = 0;                                  ///< Width in pixels
         int height = 0;                                 ///< height in pixels
-        Slang::String adapter;                          ///< Name to identify the adapter to use
-        Slang::List<Slang::String> requiredFeatures;    ///< List of required feature names. 
+        const char* adapter = nullptr;                  ///< Name to identify the adapter to use
+        int requiredFeatureCount = 0;                   ///< Number of required features.
+        const char** requiredFeatures = nullptr;        ///< Array of required feature names, whose size is `requiredFeatureCount`.
         int nvapiExtnSlot = -1;                         ///< The slot (typically UAV) used to identify NVAPI intrinsics. If >=0 NVAPI is required.
     };
 
         // Will return with SLANG_E_NOT_AVAILABLE if NVAPI can't be initialized and nvapiExtnSlot >= 0
-    virtual SLANG_NO_THROW SlangResult SLANG_MCALL initialize(const Desc& desc, void* inWindowHandle) = 0;
+    virtual SLANG_NO_THROW Result SLANG_MCALL initialize(const Desc& desc, void* inWindowHandle) = 0;
+
+    virtual SLANG_NO_THROW bool SLANG_MCALL hasFeature(const char* feature) = 0;
 
-    bool hasFeature(const Slang::UnownedStringSlice& feature) { return getFeatures().indexOf(Slang::String(feature)) != Slang::Index(-1); }
-    virtual SLANG_NO_THROW const Slang::List<Slang::String>& SLANG_MCALL getFeatures() = 0;
+        /// Returns a list of features supported by the renderer.
+    virtual SLANG_NO_THROW Result SLANG_MCALL getFeatures(const char** outFeatures, UInt bufferSize, UInt* outFeatureCount) = 0;
```

Comment:
Couldn't this be defined so that it doesn't need the `bufferSize`? That is, we could conceivably return a pointer to storage allocated when the `Renderer` was created that will have the same lifetime as the renderer, since the list of supported features is not going to change after initialization.